### ISSUE TITLE
Visualisation of svg result inside VSCode

### ISSUE THREAD

**Issue by rcqls** [davibarreira/Vizagrams.jl#40]
Hi Davi! Congrats for your excellent package which is very promising.

First, I have a very naive question: is it possible in VSCode to  visualize the svg result returned by draw in a sort of svg window? For now, I use quarto rendering (with the last julia engine) to show the results globally. I can't visualize a cells one by one (by clicking "run cell"). More generally, what's your preferred way to show the results of vizagrams.jl? (I missed that in your doc)

Also, do you plan to introduce other backends like other graphics packages do?   

Thanks again for your very nice job!

**Comment by rcqls** [comment]
Oups! Having a look in your source code, I see backends and interactive folders..... 😁

**Comment by davibarreira** [comment]
I don't use vscode much, so I'm note sure how to make it work. Do you mind sharing a screenshot so I can have a better idea?

**Comment by davibarreira** [comment]
![image](https://github.com/user-attachments/assets/f026b96b-1e44-4817-8b16-dc3aad70cdca)

Using the notebook within VSCode seems to work.

**Comment by rcqls** [comment]
> I don't use vscode much, so I'm note sure how to make it work. Do you mind sharing a screenshot so I can have a better idea?

With Plots.jl, when you use plot generic function the plot is drawn side by side.
I have tried to see if there is a vscode plugin to have side by side the result of the svg tag. Maybe saving the file first is the solution by coupling this with SVG plugin preview. I have to check that.
 
![Capture d’écran 2024-09-10 à 13 16 02](https://github.com/user-attachments/assets/678ce3db-3c79-4a88-972f-78660fd26833)

**Comment by rcqls** [comment]
> ![image](https://private-user-images.githubusercontent.com/6407557/366012575-f026b96b-1e44-4817-8b16-dc3aad70cdca.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU5NjcyOTUsIm5iZiI6MTcyNTk2Njk5NSwicGF0aCI6Ii82NDA3NTU3LzM2NjAxMjU3NS1mMDI2Yjk2Yi0xZTQ0LTQ4MTctOGIxNi1kYzNhYWQ3MGNkY2EucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkxMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MTBUMTExNjM1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODQyZGUxZDhiNTkyMjUzYWY3NzczNDE3ZDk1ZDNkNDk0Mzk3Y2MxYzE4ZWFmOTI0OGIxNGQwMzgwZmRiYjg4NiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.XVqZKT8OQs3iskn_PLt6gpRshUSUPUI_HKmYWUiUbkc)
> 
> Using the notebook within VSCode seems to work.

I guess the notebook integrated to VSCode is jupyter... I am not a fan of that because it depends on python.... I prefer quarto now with this new julia engine to be used inside vscode. For now, this engine is pretty new (the old one use jupyter and then python to render the content) and I guess a lot of evolution would come....

**Comment by davibarreira** [comment]
@rcqls , I think I got it!

I'll add this to the package. In the mean time, you can just use
```julia
function Base.show(io::IO, ::MIME"image/svg+xml", d::Union{Vizagrams.𝕋{Mark},Mark,Prim,GeometricPrimitive})
    svg_str = draw(d)
    print(io, svg_str)
end
```

![image](https://github.com/user-attachments/assets/c8a10150-bccf-4232-9e29-168c620f723b)

**Comment by davibarreira** [comment]
Let me know if this works on your side. I'm writing the second option in the package itself, i.e.:
```
function Base.show(io::IO, ::MIME"image/svg+xml",drawing::Vizagrams.Hyperscript.Node{Vizagrams.Hyperscript.HTMLSVG})
    print(io, drawing)
end
```

**Comment by davibarreira** [comment]
Yeah, I see it. It looks like something from VSCode. I'm waiting on the forum to see if anyone knows why this is happening.

**Comment by davibarreira** [comment]
The oracle has spoken:
```julia
function Base.show(io::IO, ::MIME"image/svg+xml", d::Union{𝕋{Mark},Mark,Prim,GeometricPrimitive})
    svg_str = draw(d)
    print(io, "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n")
    print(io, svg_str)
end
```

This should work!

**Comment by rcqls** [comment]
> The oracle has spoken:
> 
> ```julia
> function Base.show(io::IO, ::MIME"image/svg+xml", d::Union{𝕋{Mark},Mark,Prim,GeometricPrimitive})
>     svg_str = draw(d)
>     print(io, "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n")
>     print(io, svg_str)
> end
> ```
> 
> This should work!

I am impressed 👏.  So the panel plot expects a complete xml output....  I think it is now a great advance to have the output of each diagram built from your nice package in the vscode plot panel.

P.S.: as a slight comment not related to vscode, I am not convinced by the fact that you export `plot` generic defined in quickplot.jl...  however I understand your point eventhough it  restricts the use of Plots.jl and Vizagrams.jl in a same julia file or notebook.

**Comment by davibarreira** [comment]
@rcqls , I think I agree with you with regards of `plot`. In fact, after a refactor from version 0.1 to 0.2, the utility of `plot` is questionable. And as you say, it does not play well with other packages. I think I'll drop it! Thanks for the advice.

**Comment by rcqls** [comment]
> @rcqls , I think I agree with you with regards of `plot`. In fact, after a refactor from version 0.1 to 0.2, the utility of `plot` is questionable. And as you say, it does not play well with other packages. I think I'll drop it! Thanks for the advice.
> 
> I'm closing the issue. But feel free to reopen if needed.



> @rcqls , I think I agree with you with regards of `plot`. In fact, after a refactor from version 0.1 to 0.2, the utility of `plot` is questionable. And as you say, it does not play well with other packages. I think I'll drop it! Thanks for the advice.
> 
> I'm closing the issue. But feel free to reopen if needed.



> @rcqls , I think I agree with you with regards of `plot`. In fact, after a refactor from version 0.1 to 0.2, the utility of `plot` is questionable. And as you say, it does not play well with other packages. I think I'll drop it! Thanks for the advice.
> 
> I'm closing the issue. But feel free to reopen if needed.



> @rcqls , I think I agree with you with regards of `plot`. In fact, after a refactor from version 0.1 to 0.2, the utility of `plot` is questionable. And as you say, it does not play well with other packages. I think I'll drop it! Thanks for the advice.
> 
> I'm closing the issue. But feel free to reopen if needed.



> @rcqls , I think I agree with you with regards of `plot`. In fact, after a refactor from version 0.1 to 0.2, the utility of `plot` is questionable. And as you say, it does not play well with other packages. I think I'll drop it! Thanks for the advice.
> 
> I'm closing the issue. But feel free to reopen if needed.

Of course, it's time to close this issue and thanks again for your excellent work.... Hope version 2 would become soon since I want to introduce your package to all the people around me playing with the wondrfull julia language.

**Comment by davibarreira** [comment]
Please let me know of any "must-have" features, or difficulties/weird syntax you find with the package. I'm very eager from feedback from users.

**Comment by rcqls** [comment]
> Please let me know of any "must-have" features, or difficulties/weird syntax you find with the package. I'm very eager from feedback from users.

Hi Davi, when do you plan to offer a new version of your package. I can't tell my students to use your package if it not working on vscode 😁.

**Comment by davibarreira** [comment]
> > Please let me know of any "must-have" features, or difficulties/weird syntax you find with the package. I'm very eager from feedback from users.
> 
> 
> 
> Hi Davi, when do you plan to offer a new version of your package. I can't tell my students to use your package if it is not working on vscode 😁. 

I have already pushed the modificiation, but it seems I've forgotten to update the version.

I'll be doing it today.

**Comment by rcqls** [comment]
> > > Please let me know of any "must-have" features, or difficulties/weird syntax you find with the package. I'm very eager from feedback from users.
> > 
> > 
> > Hi Davi, when do you plan to offer a new version of your package. I can't tell my students to use your package if it is not working on vscode 😁.
> 
> I have already pushed the modificiation, but it seems I've forgotten to update the version.
> 
> I'll be doing it today.

I have already seen that you made a PR but was a bit surprised that no new minor version was added. Next thursday, I have my class about "visualization and reporting tools" so would be nice to have your package among the candidate packages for visualisation.

**Comment by davibarreira** [comment]
That is great to hear! I'm really glad to be included.
I've updated the version. In a couple of minutes it should be up.
